lkl: Make rx of virtio net edge-triggered

liuyuan10 opened the pull request:
This can avoid burning CPU when there is not enough available buffer to
receive packets.

When an incomming packet unblocks epoll_wait of rx thread and there is
no available buffer, it comes back to epoll_wait and never wakes up
(this is what edge-triggered means). But the driver will notify the
device after putting more available buffers by VIRTIO_MMIO_QUEUE_NOTIFY.
So there is no need of a busy polling for an available buffer.

Tested By:
Ran TCP_STREAM and TCP_RR 100 times.

Signed-off-by: Yuan Liu liuyuan@google.com

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/lkl/linux/149)

<!-- Reviewable:end -->

pscollins commented:
LGTM

thehajime commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/lkl/include/lkl_host.h, line 89 \[r1\]](https://reviewable.io:443/reviews/lkl/linux/149#-KFMYo5-fqbPIKzye9RN:-KFMYo5-fqbPIKzye9RO:1481475583) ([raw file](https://github.com/lkl/linux/blob/906472151b5bba613ed13df0b22d8062e6a3c820/tools/lkl/include/lkl_host.h#L89)):*
if possible, could you try to not introducing whitespace only changes to the commit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/lkl/linux/149)*
<!-- Sent from Reviewable.io -->

thehajime commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/lkl/linux/149)*
<!-- Sent from Reviewable.io -->

thehajime commented:
overall, it looks good to me.  thanks !

liuyuan10 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/lkl/include/lkl_host.h, line 89 \[r1\]](https://reviewable.io:443/reviews/lkl/linux/149#-KFMYo5-fqbPIKzye9RN:-KFNNdszcgqE3z_a1Pvg:1356055174) ([raw file](https://github.com/lkl/linux/blob/906472151b5bba613ed13df0b22d8062e6a3c820/tools/lkl/include/lkl_host.h#L89)):*
Actually I replaced the spaces with tabs which I introduced a while back carelessly. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/lkl/linux/149)*
<!-- Sent from Reviewable.io -->
